enclave/core/sgx/enter.S

```diff
@@ -151,6 +162,26 @@ oe_enter:
 
     // Start the new stack under the red zone
     sub $ABI_REDZONE_BYTE_SIZE, %r8
+
+.check_interrupt_request:
+    movq td_state(%r11), %r12
+    movq td_interrupt_pending(%r11), %r13
+    // Treat the value SIGUSR1 as a host request to interrupt
+    // the enclave
+    cmp $SIGUSR1, %rsi
```

Comment:
Can the implementation support different signal numbers? Upper level code might want to use SIGUSR1 as kickstart signal for certain operation, and also handle hostside generated SW signals, for example, socket SIGPIPE or SIGABRT in Linux.

Reply:
If the upper level code targets an in-enclave thread, it can prepare the siginfo (signo=SIGUSR1) inside the enclave and append the siginfo to target thread's signal queue. If the upper level code targets an thread running on the host, the OE host will fallback to the host-registered handler (if any).